proxy/private_rpc/pr_eth_signing_api.py

```diff
@@ -0,0 +1,33 @@
+from typing import Any, ClassVar, Dict
+
+from common.ethereum.bin_str import EthBinStrField
+from common.ethereum.hash import EthAddress
+from common.http.utils import HttpRequestCtx
+from common.neon.transaction_model import NeonTxModel
+from .server_abc import PrivateRpcApi
+
+
+class PrEthSigningApi(PrivateRpcApi):
+    name: ClassVar[str] = "PrivateRpc::EthSigning"
+
+    @PrivateRpcApi.method(name="eth_sign")
+    async def eth_sign(self, ctx: HttpRequestCtx, eth_address: str, data: str) -> str:
+        response = await self._op_client.sign_eth_message(ctx.ctx_id, eth_address, data)
+        return response.signed_message
+
+    @PrivateRpcApi.method(name="eth_signTransaction")
+    async def eth_sign_transaction(self, ctx: HttpRequestCtx, tx: Dict[str, Any]) -> EthBinStrField:
+        neon_tx = NeonTxModel.default().model_copy(
+            update={
+                "from_address": EthAddress.from_raw(tx.get("from")),
+                "to_address": EthAddress.from_raw(tx.get("to")),
+                "gas_limit": tx.get("gas", 0),
+                "gas_price": tx.get("gasPrice", 0),
+                "value": tx.get("value", 0),
+                "call_data": EthBinStrField.from_raw(tx.get("data")),
+                "nonce": tx.get("nonce", 0),
+            }
+        )
+
+        response = await self._op_client.sign_eth_transaction(ctx.ctx_id, neon_tx, tx.get("chainId", ctx.chain_id))
```

Comment:
`self.get_chain_id(ctx)`

Reply:
This now looks at the transaction's chain Id first, and if not found uses the default chain Id.